### PR TITLE
runtime: add rent debit charges to block metadata

### PR DESCRIPTION
**Pull request by t-nelson**
#### Problem

It is difficult to resolve rent debits from block metadata alone

#### Summary of Changes

Add explicit rent debit entries in block metadata

### CODE REVIEW

**Review by CriesofCarrots — APPROVED** [review]
lgtm if CI is happy

**Comment by t-nelson** [comment]
Added some tests for `RentDebits`

File: runtime/src/bank.rs [hunk 1]
```diff
@@ -3620,19 +3645,23 @@ impl Bank {
...
                 .rent_collector
                 .collect_from_existing_account(&pubkey, &mut account);
+            total_rent += rent;
             // Store all of them unconditionally to purge old AppendVec,
             // even if collected rent is 0 (= not updated).
             // Also, there's another subtle side-effect from this: this
             // ensures we verify the whole on-chain state (= all accounts)
             // via the account delta hash slowly once per an epoch.
             self.store_account(&pubkey, &account);
+            rent_debits.push(&pubkey, rent, account.lamports());
```
Comment (ryoqun) [inline runtime/src/bank.rs:3661]: hehe, nice :eyes:

**Review by mvines — APPROVED** [review]
lgtm too!

**Pull request merged**